### ISSUE TITLE
Display which database (OFF or USDA) a result is from in search results and allow searching only one

### ISSUE THREAD

**Issue by user234683** [davidhealey/waistline#526]
When searching for generic items, such as fruits, it's preferable to only check the USDA database, so as to avoid the deluge of branded items with a 100 g serving size. (see #524 ). It would be nice to have an icon or other indicator next to the result to indicate which database the result is from, and to filter which databases are being searched. This would become more useful as more databases get added.

**Comment by davidhealey** [comment]
> This would become more useful as more databases get added.

I wouldn't hold your breath that this will happen.

Where do you see this button being placed? The UI is already pretty cramped.

**Comment by user234683** [comment]
There's a filter button in the search bar already, that selects between "Tree" and "24-hour store" (Still not sure what these mean, vegan and something else?) There's room in that menu to add toggles for UDSA vs OFF.

**Comment by davidhealey** [comment]
> Tree" and "24-hour store"

@EmilJunker Any idea what this is?

**Comment by EmilJunker** [comment]
The tree and the store are simply the default food categories which I added in #421. The food categories can be customized under `Settings > Foods, Meals, Recipes > Labels and Categories`. I'm sorry if this is confusing to you. I'm open for suggestions to make this clearer.

**Comment by davidhealey** [comment]
> default food categories which I added in #421.

Oh yeah, that rings a bell.

**Comment by columndeeply** [comment]
Seconding this request, I'd love a way to only search a specific database too. Searching for basic foods and not having OFF return a bunch of stuff with that food in the title would be a huge improvement to the search functionality.

**Comment by columndeeply** [comment]
Would something like this be okay?  haven't looked at the code yet but I could give it a shot and send a pull request

![20221001_153312](https://user-images.githubusercontent.com/106948293/193412170-6f902fb0-bd88-4469-a040-9e74fcea84af.png)

**Comment by davidhealey** [comment]
It looks good

**Comment by EmilJunker** [comment]
I'm not sure if it's technically possible to implement this in the way shown in your mocup. I think we don't need to show the food categories in the filter menu when doing an online search since the search results obviously aren't categorized yet. So we can just replace the food categories with the two pseudo categories for USDA and OFF when doing an online search.

In any event, I can assure you that implementing this feature would be quite some work and require extensive code changes in a lot of the existing files (which is the reason why I have so far refrained from tackling it). But you are very welcome to try it and open a pull request.

**Comment by columndeeply** [comment]
I'll take a look this weekend, I've never worked with a Cordova app but it looks like it's mostly JS? I'll let you know if I figure out how to make this work.

**Comment by prototyperspective** [comment]
@EmilJunker @davidhealey Yes, these icons are very confusing (unclear). Could you please add some info to them such as text next to the icon or text on hover?

There could also be an introductory wizard going through the app that also explains that but while I think such a wizard would be extremely useful to make people learn the basics quickly, this would probably be too detailed (too much to learn at once and taking too much time). Alternatively, just disable all the default categories. I still haven't figured out what these things are, specifically which items are contained in each – I thought the tree contained raw and/or generic foods and the latter contained anything else but that doesn't seem to be the case. Maybe there should be a separate issue for this.
